### PR TITLE
fix: Promise polyfill for zone

### PR DESCRIPTION
**Pull request by patrickhulce**
R: @brendankenny 

Fixes evaluateAsync behavior when [zone.js](https://github.com/angular/zone.js) has mucked with Promise.
Adds zone.js to the smoke tests so regression won't happen again.

Addresses #1173.

### CODE REVIEW

File: lighthouse-core/gather/driver.js [hunk 1]
```diff
@@ -143,15 +143,19 @@ class Driver {
       );
 
       this.sendCommand('Runtime.evaluate', {
-        // We need to wrap the raw expression for several purposes
+        // We need to expliticly wrap the raw expression for several purposes:
```
Comment (ebidel) [inline lighthouse-core/gather/driver.js]: explicitly

**Comment by brendankenny** [comment]
> Should we bring in zones via npm?

Looks like you didn't have to modify @patrickhulce, so this could work this time?

**Comment by patrickhulce** [comment]
yeah true I didn't this time around, will fix

File: lighthouse-core/gather/driver.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/gather/driver.js]: oops, done :)

File: lighthouse-cli/test/fixtures/static-server.js [hunk 2]
```diff
@@ -27,11 +27,11 @@ function requestHandler(request, response) {
...
   let absoluteFilePath = path.join(__dirname, filePath);
 
-  if (filePath === '/promise_polyfill.js') {
+  if (filePath === '/zone.js') {
     // evaluateAsync previously had a bug that LH would fail if a page polyfilled Promise.
-    // We bring in a third-party Promise polyfill to ensure we don't still fail.
-    const thirdPartyPath = '../../../lighthouse-core/third_party';
-    absoluteFilePath = path.join(__dirname, `${thirdPartyPath}/promise-polyfill/promise.js`);
+    // We bring in an aggressive Promise polyfill (zone) to ensure we don't still fail.
+    const zonePath = '../../../node_modules/zone.js';
```
Comment (ebidel) [inline lighthouse-cli/test/fixtures/static-server.js]: '../../../node_modules/`

Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/static-server.js]: @ebidel not sure what you're requesting here, to change zonePath to nodeModule path?

Comment (ebidel) [inline lighthouse-cli/test/fixtures/static-server.js]: OH nvm. This is probably where it gets installed. Just looks funny from a reviewer's perspective :)

Comment (ebidel) [inline lighthouse-cli/test/fixtures/static-server.js]: All 👍

File: lighthouse-core/gather/driver.js [hunk 3]
```diff
@@ -143,15 +143,19 @@ class Driver {
...
           const __nativePromise = window.__nativePromise || Promise;
-          return __nativePromise.resolve()
-            .then(_ => ${expression})
-            .catch(${wrapRuntimeEvalErrorInBrowser.toString()});
+          return new __nativePromise(function (resolve) {
+            return __nativePromise.resolve()
+              .then(_ => ${expression})
+              .catch(${wrapRuntimeEvalErrorInBrowser.toString()})
+              .then(resolve);
+          });
```
Comment (brendankenny) [inline lighthouse-core/gather/driver.js]: I tried a few permutations of this and it looks like this is about the only decent one that works :) Reason in https://github.com/GoogleChrome/lighthouse/issues/1173#issuecomment-268107877.

For anyone following along, the nice thing about this form is that it lets the polyfill handle resolving to a value (or error), and then all we have to assume is that our `__nativePromise` constructor is intact, not our earlier assumption that its prototype chain had also not been altered

**Pull request merged**